nexus/types/src/external_api/params.rs

```diff
@@ -1021,6 +1021,10 @@ pub struct InstanceCreate {
     #[serde(default)]
     pub disks: Vec<InstanceDiskAttachment>,
 
+    /// Choice of which disk this instance should boot from.
+    #[serde(default)]
+    pub boot_device: Option<String>,
+
```

Comment:
Can this be something other than a disk, or will it be in the future? If not it would more concrete as `boot_disk`.

Reply:
it technically _can_ be a network interface, but i haven't tested that at all. there's support for iPXE in OVMF, but i'm not sure what we're building into that currently, and i'm not sure that's the way we'd _want_ to support network booting in the future.

so realistically this is a similar "how future-proof should this be" question: `boot_disk` makes sense today and probably for a while, but in the future i can imagine a list of boot options that are either disks or network interfaces, some of which with additional settings on them. in user-facing APIs do we scope this down to what we know now and imminently, or do we hedge for the future options too?

Reply:
Note also that the naming convention would be to call it `boot_disk` here when it’s a name or ID, but `boot_disk_id` on the response when it’s always an ID.

Reply:
this is now `boot_disk`, and the field on the response is `boot_disk_id`: https://github.com/oxidecomputer/omicron/pull/6585/commits/4eb1ff58d9b884f2294633b24c0d7c0e9a44da59

Reply:
How does this interact with the `disks` field? I expect this to be a `InstanceDiskAttachment` but I confess that I haven't thought about it too hard.

Reply:
Distinguishing between creates and attaches isn’t necessary because either way, you go and match up this name or ID with one of the items in the list of disks and there can be no ambiguity whether it’s by name or ID. The only quirk, which I ran into doing the console mock API, is that you can’t designate a disk you’re creating as boot disk by ID because it doesn’t have an ID yet.

Reply:
Another reason is that currently `InstanceDiskAttachment` only lets you attach by name. So you would not be able to use an ID here.

https://github.com/oxidecomputer/omicron/blob/f21d79dd2f86a814bf02569e5142621ba119e843/nexus/types/src/external_api/params.rs#L938-L953

But this argument is weakened by the fact that you already have to have the name on hand to attach the disk. That said, I'd rather this be correct already (use `NameOrId`) and consistent with the `PUT` endpoint.

Reply:
also one of the next things i want to do is move `InstanceDiskAttachment` to take `NameOrId`. so i lean towards making this is more like the future we'd move towards.